sale_stock_line_customer_ref/models/stock_move_line.py

```diff
@@ -20,11 +20,13 @@ def write(self, vals):
     @contextmanager
     def update_related_packages(self, vals):
         """Keep packages in sync."""
-        # FIXME: is not granted that all the lines involved by the pkg will be updated here.
+        # FIXME: is not granted that all the lines involved
+        #  by the pkg will be updated here.
         # We should look for all lines linked to the same picking and update them all.
         result_package_updated = "result_package_id" in vals
+        old_packages = self.env["stock.quant.package"]
         if result_package_updated:
-            old_packages = self.result_package_id
+            old_packages |= self.result_package_id
```

Comment:
Why?

Reply:
```suggestion
        if result_package_updated:
            old_packages = self.result_package_id
```

Reply:
I added the `old_packages` variable declaration because it is currently used inside the if block below, but not declared outside so there will be a slight warning like: `Local variable 'old_packages' might be referenced before assignment`

Reply:
I don't see any such notification

Reply:
Some code editors will ignore it, but logically it can be seen that `old_packages` is not declared but assigned in an `if` block and used in a peer `if` block below

![image](https://github.com/user-attachments/assets/418a395c-3053-4392-acfe-c8c1e5ca3025)

Reply:
We will be interested when there is a WARNING displayed on the terminal. The debug parts supported by the EDITOR are only general and follow available templates.